tests/test_on_data/test_imaging_interfaces.py

```diff
@@ -75,20 +77,26 @@ def check_extracted_metadata(self, metadata: dict):
         )
 
 
+@parameterized_class(
+    [
+        {
+            "interface_kwargs": {
+                "folder_path": str(OPHYS_DATA_PATH / "imaging_datasets" / "ScanImage"),
+                "file_pattern": "scanimage_20240320_multifile*.tif",
+                "channel_name": "Channel 1",
+            }
+        },
+        {
+            "interface_kwargs": {
+                "folder_path": str(OPHYS_DATA_PATH / "imaging_datasets" / "ScanImage"),
+                "file_pattern": "scanimage_20240320_multifile*.tif",
+                "channel_name": "Channel 2",
+            }
+        },
+    ]
+)
```

Comment:
@pauladkisson tried out your suggestion but I got this error:
```python-traceback
src/neuroconv/tools/testing/data_interface_mixins.py:105 (TestScanImageSinglePlaneMultiFileImagingInterface.test_conversion_as_lone_interface)
self = <tests.test_on_data.test_imaging_interfaces.TestScanImageSinglePlaneMultiFileImagingInterface testMethod=test_conversion_as_lone_interface>

    def test_conversion_as_lone_interface(self):
>       interface_kwargs = self.interface_kwargs
E       AttributeError: 'TestScanImageSinglePlaneMultiFileImagingInterface' object has no attribute 'interface_kwargs'
```
is there something that has to be done differently for `interface_kwargs` compared to `conversion_options` to make this work?

Reply:
OK, I tracked down what's going wrong here. When you use `parameterized.parameterize_class` on a test case with Mixins, pytest will instantiate a class object for each parameter --> run all tests for each test class object AND instantiate an unparameterized class object and run only the Mixin tests for that object.  See dummy example below:
```python
class DebugMixin:
    def test_mixin(self):
        assert self.a

@parameterized_class(
    [
        {'a' : 1},
        {'a' : 2},
    ],
)
class TestDebug(TestCase, DebugMixin):
    def test_debug(self):
        assert self.a
```

This will generate the following output:
```bash
pytest tests/test_on_data/test_imaging_interfaces.py -k TestDebug -s --verbose                                                                                                                      1 ↵
============================================================================================ test session starts ============================================================================================
platform darwin -- Python 3.9.18, pytest-8.0.0, pluggy-1.4.0 -- /opt/anaconda3/envs/neuroconv_env/bin/python
cachedir: .pytest_cache
rootdir: /Users/pauladkisson/Documents/CatalystNeuro/NWB/neuroconv
configfile: pyproject.toml
plugins: cov-4.1.0
collecting ... Running GIN tests locally!
collected 61 items / 56 deselected / 5 selected                                                                                                                                                             

tests/test_on_data/test_imaging_interfaces.py::TestDebug_0::test_debug PASSED
tests/test_on_data/test_imaging_interfaces.py::TestDebug_0::test_mixin PASSED
tests/test_on_data/test_imaging_interfaces.py::TestDebug_1::test_debug PASSED
tests/test_on_data/test_imaging_interfaces.py::TestDebug_1::test_mixin PASSED
tests/test_on_data/test_imaging_interfaces.py::TestDebug::test_mixin FAILED

================================================================================================= FAILURES ==================================================================================================
___________________________________________________________________________________________ TestDebug.test_mixin ____________________________________________________________________________________________

self = <tests.test_on_data.test_imaging_interfaces.TestDebug testMethod=test_mixin>

    def test_mixin(self):
>       assert self.a
E       AttributeError: 'TestDebug' object has no attribute 'a'

tests/test_on_data/test_imaging_interfaces.py:81: AttributeError
============================================================================================= warnings summary ==============================================================================================
../../../../../../opt/anaconda3/envs/neuroconv_env/lib/python3.9/site-packages/hdmf/container.py:11
  /opt/anaconda3/envs/neuroconv_env/lib/python3.9/site-packages/hdmf/container.py:11: DeprecationWarning: 
  Pyarrow will become a required dependency of pandas in the next major release of pandas (pandas 3.0),
  (to allow more performant data types, such as the Arrow string type, and better interoperability with other libraries)
  but was not found to be installed on your system.
  If this would cause problems for you,
  please provide us feedback at https://github.com/pandas-dev/pandas/issues/54466
          
    import pandas as pd

src/neuroconv/tools/yaml_conversion_specification/_yaml_conversion_specification.py:7
src/neuroconv/tools/yaml_conversion_specification/_yaml_conversion_specification.py:7
  /Users/pauladkisson/Documents/CatalystNeuro/NWB/neuroconv/src/neuroconv/tools/yaml_conversion_specification/_yaml_conversion_specification.py:7: DeprecationWarning: jsonschema.RefResolver is deprecated as of v4.18.0, in favor of the https://github.com/python-jsonschema/referencing library, which provides more compliant referencing behavior as well as more flexible APIs for customization. A future release will remove RefResolver. Please file a feature request (on referencing) if you are missing an API for the kind of customization you need.
    from jsonschema import RefResolver, validate

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
========================================================================================== short test summary info ==========================================================================================
FAILED tests/test_on_data/test_imaging_interfaces.py::TestDebug::test_mixin - AttributeError: 'TestDebug' object has no attribute 'a'
```

Notice how the mix in test is run one extra time with no `a` parameter, which causes the test to fail.

Reply:
I see now, thank you for the explanation!